diff --git a/fast_tokenizer/cmake/external/gflags.cmake b/fast_tokenizer/cmake/external/gflags.cmake
index df5b3642..fcf385d8 100644
--- a/fast_tokenizer/cmake/external/gflags.cmake
+++ b/fast_tokenizer/cmake/external/gflags.cmake
@@ -23,8 +23,8 @@ IF(WIN32)
   set(GFLAGS_LIBRARIES "${GFLAGS_INSTALL_DIR}/lib/gflags_static.lib" CACHE FILEPATH "GFLAGS_LIBRARIES" FORCE)
 ELSE(WIN32)
   set(GFLAGS_LIBRARIES "${GFLAGS_INSTALL_DIR}/lib/libgflags.a" CACHE FILEPATH "GFLAGS_LIBRARIES" FORCE)
-  set(BUILD_COMMAND $(MAKE) --silent)
-  set(INSTALL_COMMAND $(MAKE) install)
+  set(BUILD_COMMAND ${CMAKE_COMMAND} --build .)
+  set(INSTALL_COMMAND ${CMAKE_COMMAND} --install .)
 ENDIF(WIN32)
 
 INCLUDE_DIRECTORIES(${GFLAGS_INCLUDE_DIR})
